### ISSUE TITLE
apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated

### ISSUE THREAD

**Issue by ksstott** [Azure/aad-pod-identity#922]
**Have you**

- [x] Read [Troubleshooting Guide](../../docs/readmes/README.troubleshooting.md)
- [x] Read [Known Issues](../../docs/readmes/README.known-issues.md)
- [x] Searched on [GitHub issues](https://github.com/Azure/aad-pod-identity/issues)

**Describe the bug**
When installing aad-pod-identity the following warnings display:
```
W0104 17:14:45.789938    9304 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0104 17:14:45.789938    9304 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0104 17:14:45.806949    9304 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0104 17:14:45.807945    9304 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0104 17:14:47.915382    9304 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0104 17:14:47.946916    9304 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0104 17:14:47.981923    9304 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0104 17:14:48.017915    9304 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
```

**Steps To Reproduce**
1. Deploy aad-pod-identity using [instructions](https://azure.github.io/aad-pod-identity/docs/getting-started/installation). I used helm for installation but I imagine any method will incur the same problem.
2. Observe warning messages

**Expected behavior**
No warning messages about deprecated versions

**AAD Pod Identity version**
1.7.1

**Kubernetes version**
1.19.3

**Additional context**
I'd be happy to see about updating these yaml files to use the latest api and getting a PR raised but I'll need to time to get my head around the contributing guide 😃

**Comment by aramase** [comment]
@ksstott We already have an issue open for this: https://github.com/Azure/aad-pod-identity/issues/661. Helm 3 CRDs dir doesn't provide an option to template. Updating to `apiextensions.k8s.io/v1` will make the helm charts unusable for Kubernetes version < 1.16. So as posted in the issue we'll be updating the api version for CRDs in the future releases once users are moving to 1.16+ as default.

**Comment by ksstott** [comment]
Ah thanks for the quick reply, I had tried looking for an issue as it seemed unlikely that it wasn't already raised but never found it. (I must have only searched for `apiextensions.k8s.io/v1beta` rather than `apiextensions.k8s.io/v1`).
That answer makes sense, I'll close this down as it's being addressed in #661.
